### DISCUSSION TITLE
high latency

### DISCUSSION THREAD

**Discussion by yashamon** [helix-editor/helix#11865] [General]
With empty config. OS windows 11. Windows Terminal. Latency measured with typometer.

The latency with an empty buffer is low around 15 ms average. However the latency jumps to around 60 average with high of 90 ms, with non empty buffer.  

For comparison under the same conditions the latency in neovim is about 30 average. 20 average when using a gui like nvy. 
Similarly with vscode.

**Comment by TornaxO7** [comment]
Hmm... may I ask how you installed helix and which terminal you are using?

**Comment by kirawi** [comment]
It's probably related to the terminal emulator per https://github.com/helix-editor/helix/discussions/9442

**Comment by yashamon** [comment]
@TornaxO7 Installed via scoop. Windows terminal.

**Comment by yashamon** [comment]
> It's probably related to the terminal emulator per #9442

@kirawi I doubt it is the terminal, the latency disappears if the  buffer is empty.
It is probably tree-sitter implementation, since the latter parses on each key stroke.

**Reply by kirawi** [reply]
Tree-sitter is not used on plain text buffers.

**Reply by yashamon** [reply]
> Tree-sitter is not used on plain text buffers.

I never mentioned plain text, I said non-empty buffer. To be more specific I was testing with a latex document.

**Reply by kirawi** [reply]
The set of all non-empty buffers includes plain text buffers, so it would have invalidated the theory of tree-sitter being the cause hence why I highlighted it.

You should test on plain text as well to rule it out. Are you also using `nvim-treesitter` for the same document?

**Reply by yashamon** [reply]
> The set of all non-empty buffers includes plain text buffers, so it would have invalidated the theory of tree-sitter being the cause hence why I highlighted it.
> 
> You should test on plain text as well to rule it out. Are you also using `nvim-treesitter` for the same document?

You could have communicated better that you wish for me to test plain text. Yes with plain text latency goes back to normal. Tested on a large plain text file.

**Reply by kirawi** [reply]
It wasn't that I wanted you to test. I assumed you already did, otherwise you would have specified _Latex_ code. Again with neovim are you using `nvim-treesitter`?

**Reply by yashamon** [reply]
> It wasn't that I wanted you to test. I assumed you already did, otherwise you would have specified _Latex_ code. Again with neovim are you using `nvim-treesitter`?

Wow that is some assumption. You are assumed I was sure that it was tree-sitter, that I knew that tree-sitter does not parse plain text, that have thought to test it, I guess you also assumed that I was a tree-sitter developer and not some random academic. 

I mentioned tree-sitter as a conjecture, which is now apparently verified so let's stick to the issue. Is there anything more required of me? I would rather not interact with you again.

**Reply by kirawi** [reply]
I was simply reminding you to be mindful of how you communicate. But I'll defer this to one of the other members of the team.

**Reply by yashamon** [reply]
> Again with neovim are you using `nvim-treesitter`? 

Yes, in fact I am getting around 30 latency with a full config of a bunch of plugins.
In case of helix the latency is around 60 with a clean state, no config.

**Comment by yashamon** [comment]
Why is it converted to a discussion?  The issue is unlikely to be a feature, so is probably a bug.

**Reply by the-mikedavis** [reply]
We convert most issues to discussions that are not immediately actionable. This needs further exploration to determine why latency is high according to typometer and discussions are a better format for triage/troubleshooting/exploration
